contracts/modules/utils/RequireUtils.sol

```diff
@@ -26,15 +33,24 @@ contract RequireUtils {
     bytes _signers
   );
 
+  event RequiredSigner(
+    address indexed _wallet,
+    address indexed _signer
+  );
+
+  mapping(address => uint256) public lastSignerUpdate;
+  mapping(address => uint256) public lastWalletUpdate;
+
   constructor(address _factory, address _mainModule) public {
     FACTORY = _factory;
     INIT_CODE_HASH = keccak256(abi.encodePacked(Wallet.creationCode, uint256(_mainModule)));
   }
 
-  function requireConfig(
+  function publishConfig(
```

Comment:
Should we deprecate this function? It's not secure against poison attack so might as well not use it.

Reply:
I guess we would only use it for when we add a new signer just to update the `lastWalletUpdate` value?

Reply:
It's is secure, it only publishes a configuration for a given imageHash and wallet, it doesn't mean you control the wallet or any of it's signers, it's just that you know what's the configuration. (That's why it doesn't emit any `RequiredSigner` events)

Reply:
@Agusx1211 could you include some comments above this function about its intended use, describe the warning PhABC brought up, and mention why it's not a concern of poisoning etc.? Very helpful details to include so the code is readable and design is described at a function-level